src/utils/document-helpers.ts

```diff
@@ -470,3 +475,20 @@ export const prepareOperations = (
     result.integrityIssues.push(...integrityErrors);
     return result;
 };
+
+export function removeExistingOperations(
+    newOperations: Operation[],
+    operationsHistory: Operation[]
+): Operation[] {
+    return newOperations.filter(newOperation => {
+        return !operationsHistory.some(historyOperation => {
+            return (
+                newOperation.index === historyOperation.index &&
+                newOperation.skip === historyOperation.skip &&
+                newOperation.scope === historyOperation.scope &&
```

Comment:
Should we also compare the branch?

Reply:
mmm I don't think so, this one is meant to be used only with operations that belongs to the same branch, maybe that filter should happen before.
right now this function is only used when you try to incorporate new operations into your document history (same branch).